### PR TITLE
Evaluatorinterface

### LINKED ISSUE #283: Create "Evaluator" interface.
Generally, our evaluators are pretty simple - they take in two datasets of a particular type (predictions, labels) and output some score. They are parameterized differently (e.g. num classes, etc.) but this could be a constructor parameter.

With the advent of `PipelineDataset`, doing the "right thing" based on inputs being dataset/dataset, rdd/rdd, or a mix of both is boilerplate logic that we should only write down once. 

I propose a `trait Evaluator` that looks something like this:

```
trait Evaluator[P,L,E] {
  def apply(predictions: RDD[P], labels: RDD[L]): E
  def apply(predictions: PipelineDataset[P], labels: RDD[L]) = apply(predictions.get, labels)
  def apply(predictions: RDD[P], labels: PipelineDataset[L]) = apply(predictions, labels.get)
  def apply(predictions: PipelineDataset[P], labels: PipelineDataset[L]) = apply(predictions.get, labels.get)
}
```

Other options include doing something clever with implicits, but I feel like this is more explicit.

### PR DESCRIPTION
**Pull request by etrain**
Closes #283.

### CODE REVIEW

File: src/main/scala/evaluation/Evaluator.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package evaluation
+
+import org.apache.spark.rdd.RDD
+import workflow.PipelineDataset
+
+trait Evaluator[P,L,E] {
```
Comment (tomerk) [inline src/main/scala/evaluation/Evaluator.scala]: I actually posted this comment on friday, but I forgot to press 'submit review' so I guess it didn't post until now...

Comment (etrain) [inline src/main/scala/evaluation/Evaluator.scala]: Fixed.

Comment (shivaram) [inline src/main/scala/evaluation/Evaluator.scala]: Might want to add scaladoc here ?

File: src/main/scala/evaluation/MeanAveragePrecisionEvaluator.scala [hunk 2]
```diff
@@ -7,23 +7,24 @@ import org.apache.spark.SparkContext._
...
   /**
    * Compute the mean average precision for multi-class classification
    * NOTE: The current implementation is only suitable when we have a 
    *       small number of classes and data items.
-   * @param testActual - For every test image, this contains list of valid labels. 
+   * @param testActual For every test image, this contains list of valid labels.
    *                     Labels are assumed to be class ids.
-   * @param testPredicted - For every test image, this contains a list of scores for each class
-   * @param numClasses - Total number of classes
+   * @param testPredicted For every test image, this contains a list of scores for each class
```
Comment (shivaram) [inline src/main/scala/evaluation/MeanAveragePrecisionEvaluator.scala]: nit - swap the order of doc here to match argument order ?

**Pull request merged**